spec/controllers/todos_controller_spec.rb

```diff
@@ -0,0 +1,88 @@
+require "rails_helper"
+include SessionsHelper
+
+RSpec.describe TodosController, type: :controller do
+  let(:user) { create(:confirmed_user) }
+
+  before do
+    # log the user in
+    session[:user_id] = user.id
+  end
+
+  describe "POST #create" do
+    let(:todo)             { build(:todo) }
+    let(:context)          { double(:context, success?: true, user: user) }
+    let(:todo_params)      { { description: todo.description } }
+    let(:interactor_input) { { user: current_user, todo_params: todo_params } }
+    let(:params)           { { todo: todo_params, user_id: user.id } }
+
+    before(:example) do
+      allow(CreateTodo).to receive(:call).with(interactor_input).
+        and_return(context)
+    end
+
+    context "when successful" do
+      it "calls the CreateTodo interactor" do
+        expect(CreateTodo).to receive(:call)
+        post :create, params
+      end
+
+      it "returns http status 302" do
+        post :create, params
+        expect(response).to have_http_status(302)
+      end
+
+      it "redirects to user profile" do
+        expect(post :create, params).to redirect_to(user)
+      end
+    end
+
+    context "when unsuccessful" do
+      before do
+        todo.description = ""
+      end
+
+      it "redirects to user profile" do
+        expect(post :create, params).to redirect_to(user)
+      end
+    end
+  end
+
+  describe "DELETE #destroy" do
+    let(:todo)             { create(:todo, user_id: user.id) }
+    let(:params)           { { id: todo.id } }
+    let(:interactor_input) { { todo_id: params.fetch(:id).to_s } }
+
+    let(:context) do
+      double(:context, success?: true, user: user, todo: todo)
+    end
+
+    before(:example) do
+      allow(DestroyTodo).to receive(:call).with(interactor_input).
+        and_return(context)
+    end
+
+    context "when successful" do
+      it "calls the DestroyTodo interactor" do
+        expect(DestroyTodo).to receive(:call)
+        delete :destroy, params
+      end
+    end
+
+    context "when when html response is requested" do
+      it "renders html" do
+        delete :destroy, params
+        expect(response.headers["Content-Type"]).
+          to eq "text/html; charset=utf-8"
+      end
+    end
+
+    context "when js response is requested" do
+      it "renders js" do
+        xhr :delete, :destroy, params
```

Comment:
Hey @npauzenga! `xhr` works most like the other controller methods `get`, `put`, `post`, etc. The difference is that it sets the appropriate headers and parameters on the requests so that Rails understands it as an AJAX request :smile: 

`xhr` stands for `XmlHttpRequest`, which I think is very confusing given that you are not using XML but I'm sure there's some story behind it :stuck_out_tongue: